src/topography.f90

```diff
@@ -72,6 +75,19 @@ type(topography_t) function topography_constructor(filename) result(topog)
     call handle_error(nf90_get_att(ncid, depth_id, 'maximum_depth', topog%max_depth), isfatal=.false.)
     call handle_error(nf90_get_att(ncid, depth_id, 'nonadvective_cells_removed', topog%nonadvective_cells_removed), isfatal=.false.)
 
+    if (present(grid_type)) then
+      topog%grid_type = grid_type ! grid_type arg overrides value in file
+    else
+      call handle_error(nf90_get_att(ncid, depth_id, 'grid_type', file_grid_type), isfatal=.false.)
+      if ( file_grid_type == 'B' .or. file_grid_type == 'C' ) then
+        topog%grid_type = file_grid_type
+      end if
+    end if
+    write(output_unit,*) "  grid_type = ", topog%grid_type
+    if (.not.( topog%grid_type == 'B' .or. topog%grid_type == 'C' )) then
```

Comment:
```suggestion
    if (all(topog%grid_type /= ['B', 'C'])) then
```
Totally optional, but I find it more compact and easier to read.

Reply:
thanks, that's much nicer